### DISCUSSION TITLE
Add custom exception handling class per APIView

### DISCUSSION THREAD

**Discussion by lipka** [encode/django-rest-framework#7859] [Ideas & Suggestions]
Would it be possible to add custom exception handling per each APIView? It could just follow the same style[1] as all the other classes and just fallback to the original behavior if not set.

[1] https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/views.py#L101

**Comment by xordoquy** [comment]
You can simply override the `handle_exception` on a per view basis (https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/views.py#L413).

**Comment by delinhabit** [comment]
@xordoquy That is indeed possible, but then you have to override (and possible copy+paste) the entire method just to change the particular `exception_handler`. Doesn't feel like a very DRY approach.

Would you consider having this as a feature if I can come up with a PR?

**Comment by lovelydinosaur** [comment]
Not something we'd support no. The exception handler _should_ be a global case. For more specific cases write the view logic explicitly.

**Comment by mrdaliri** [comment]
In recent versions, you can override `get_exception_handler` and return your custom handler for a specific APIView/ViewSet. See PR #4753 for more info.

https://github.com/encode/django-rest-framework/blob/b25ac6c5e36403f62b13163a0190eaa48b586c47/rest_framework/views.py#L294-L298
